extensions/common/context.cc

```diff
@@ -62,11 +62,40 @@ void extractFqdn(const std::string& cluster_name, std::string* fqdn) {
   }
 }
 
-// Extract service name from service fqdn.
-void extractServiceName(const std::string& fqdn, std::string* service_name) {
-  const std::vector<std::string>& parts = absl::StrSplit(fqdn, '.');
-  if (parts.size() > 0) {
-    *service_name = parts[0];
+// Extract service name from service host.
+void extractServiceName(const std::string& host, std::string* service_name,
```

Comment:
how is this logic portable for non k8s platforms or for external services that are added via service entries?

Reply:
Mixer v1 uses host as service name for external service, so the logic here indeed breaks parity. Is there a good indicator of external service cluster?

Reply:
Yes there is. We already know the destination namespace, if we do not know the destination namespace,, ie. metadata exchange was not sent by the other side, then it is an external service.

So we should use the full service host. It does bring into question how we choose to use service short name. alternatively, there *is no* short name for external services.

@kyessenov @douglas-reid what do you think about not having service name at all for external services?

Reply:
Sorry resolved by mistake. What happens in the egress gateway? Are we going to have different hosts for same gateway service ?

Reply:
@mandarjog I think we need a before / after of the telemetry for an external service -- and we should probably run that by the WG.

Reply:
I've added more comments how the external service name is decided. It should have achieved parity with the Mixer v1. Ptal. Thanks!